### ISSUE TITLE
How to use unlabeled data?

### ISSUE THREAD

**Issue by kun-dragon** [Mumuqiao/MixCycle#1]
Good works! 
After reading the paper, there is still a question about 'how to use the unlabeled data'. Could you explain clearly?

**Comment by Mumuqiao** [comment]
Thank you for your interest!
We construct point cloud sequence {P1, P2, P3, ..., Pn} for training, and P2 to Pn are the unlabeled data. 
We first track forward the PC sequence from P1 to Pn and get results bonding box B1 to Bn.
Then we track backward the PC sequence from Pn to P1 and get results B\`n to B\`1.
For the unlabeled data, we measure the consistencies between Bi and B\`i as loss.

**Comment by kun-dragon** [comment]
In your paper, n is set to 2, meaning only P2 is unlabeled data and P1 is labeled data. So there are 50% training data labeled. How to train with 1% or 10% training data?

**Comment by Mumuqiao** [comment]
In our paper, we start the point cloud sequence from P0. So P1 and P2 are the unlabeled data.
We use labeled data to construct training sequences. So if we have 2 labeled data, we can only construct 2 PC sequences for training.

**Comment by kun-dragon** [comment]
P0 is labeled, P1 and P2 are unlabeled data. Does that mean that 1/3 of training data are labeled?  Is the sampling rate 1/3?  
As for the 10% sampling rate, a sequence consists of P0, P1, P2,  ..., and P9, where only P0 is labeled. Is that right?

**Comment by Mumuqiao** [comment]
The sampling rate represents the number of labels we use as a proportion of the number of labels in the entire dataset.
It does not represent the percentage of the number of labels in the length of the training PC sequence.
So, for example, 100 labels total in the dataset.  We can get only 1 label and 1 training PC sequence P0 P1 P2 with 1% sampling rate. Now, we are using 1 labeled data and 2 unlabeled data.
I guess you calculated the sampling rate using 1/(1+2) instead of 1/100.

**Comment by kun-dragon** [comment]
I got it. In KITTI, Nusences, and Waymo, each point cloud frame has several labeled objects, so the sampling rate is not related to the percentage of the number of labels in the length of the training PC sequence. However, as for 2D SOT data(for example, GOT10k), each frame image has only one labeled data, so the sampling rate is equal to that percentage of the number of labels in the length of the training sequence.

**Comment by Mumuqiao** [comment]
OK~ Now I know why you are confused about it. Indeed it's quite different with 2D SOT.

**Comment by kun-dragon** [comment]
Thanks a lot!
